File: js/src/gdb/tests/test-JSSymbol.py
# Printing JS::Symbols.

assert_subprinter_registered('SpiderMonkey', 'ptr-to-JS::Symbol')

run_fragment('JSSymbol.simple')

assert_pretty('unique', 'Symbol()')
assert_pretty('unique_with_desc', 'Symbol("Hello!")')
assert_pretty('registry', 'Symbol.for("Hello!")')
assert_pretty('well_known', 'Symbol.iterator')

File: js/src/gdb/tests/test-JSSymbol.cpp
#include "gdb-tests.h"

#include "js/String.h"
#include "js/Symbol.h"

FRAGMENT(JSSymbol, simple) {
  using namespace JS;

  RootedString hello(cx, JS_NewStringCopyZ(cx, "Hello!"));

  Rooted<Symbol*> unique(cx, NewSymbol(cx, nullptr));
  Rooted<Symbol*> unique_with_desc(cx, NewSymbol(cx, hello));
  Rooted<Symbol*> registry(cx, GetSymbolFor(cx, hello));
  Rooted<Symbol*> well_known(cx, GetWellKnownSymbol(cx, SymbolCode::iterator));

  breakpoint();

  use(unique);
  use(unique_with_desc);
  use(registry);
  use(well_known);
}
